org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java

```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
 			format(ent);
 	}
 
+	/**
+	 * Format the diff entries by filtering out the noise from the given delta filter pattern
+	 * The filter acts only for files that have MODIFY change Type.
+	 * If there are no changes detected, we will remove the diff entry.
+	 * @param entries
+	 * @param deltaFilterPattern
+	 * @throws IOException
+	 */
+
+	public void format(List<? extends DiffEntry> entries, Pattern deltaFilterPattern) throws IOException {
+		if (deltaFilterPattern == null)
+			format(entries);
+		else {
+
+			Iterator<? extends DiffEntry> diIterator = entries.iterator();
+
+			while (diIterator.hasNext()) {
+
+				DiffEntry diffEntry = diIterator.next();
+
+				if (MODIFY.equals(diffEntry.changeType)) {
```

Comment:
Don't need newlines between ALL of these lines... Makes the code harder to read.

Reply:
:)

Reply:
changed it